lightstep/resource_opsgenie_destination_test.go

```diff
@@ -0,0 +1,150 @@
+package lightstep
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"testing"
+
+	"github.com/lightstep/terraform-provider-lightstep/client"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestAccOpsgenieDestination(t *testing.T) {
+	var destination client.Destination
+
+	missingUrlConfig := `
+resource "lightstep_opsgenie_destination" "missing_url" {
+  project_name = ` + fmt.Sprintf("\"%s\"", testProject) + `
+  destination_name = "my-destination"
+  auth {
+    username = ""
+    password = "pass123"
+  }
+}
+`
+	missingAuthConfig := `
+resource "lightstep_opsgenie_destination" "missing_auth" {
+  project_name = ` + fmt.Sprintf("\"%s\"", testProject) + `
+  destination_name = "my-destination"
+  url = "https://example.com"
+}
+`
+
+	destinationConfig := `
+resource "lightstep_opsgenie_destination" "opsgenie" {
+  project_name = ` + fmt.Sprintf("\"%s\"", testProject) + `
+  destination_name = "my-destination"
+  url = "https://example.com"
+  auth {
+    username = ""
+    password = "pass123"
+  }
+}
+`
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccOpsgenieDestinationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: missingUrlConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckOpsgenieDestinationExists("lightstep_opsgenie_destination.missing_url", &destination),
+				),
+				ExpectError: regexp.MustCompile("The argument \"url\" is required, but no definition was found."),
+			},
+			{
+				Config: missingAuthConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckOpsgenieDestinationExists("lightstep_opsgenie_destination.missing_auth", &destination),
+				),
+				ExpectError: regexp.MustCompile("Insufficient auth blocks"),
+			},
+			{
+				Config: destinationConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckWebhookDestinationExists("lightstep_opsgenie_destination.opsgenie", &destination),
+					resource.TestCheckResourceAttr("lightstep_opsgenie_destination.opsgenie", "destination_name", "my-destination"),
+					resource.TestCheckResourceAttr("lightstep_opsgenie_destination.opsgenie", "url", "https://example.com"),
+					resource.TestCheckResourceAttr("lightstep_opsgenie_destination.opsgenie", "auth.0.username", ""),
```

Comment:
what happens if they do specify a username? is that an option we even want to expose if it will never be needed?

Reply:
I waffled on this. On one hand opsgenie is accepting a basic auth token so it makes sense to support username if they ever start using it. On the other hand we could hide this from the user in terraform just like we do in the UI. (It wouldn't be hidden in the public API though.

Reply:
Do you have a preference?

Reply:
I'd vote to hide it in terraform, assuming that doesn't create more work for you, in the interest of making terraform as minimally confusing as possible.